### PR TITLE
Add support for adding CSP nonce to profile handler JS script tag

### PR DESCRIPTION
**Pull request by richhollis**

### CODE REVIEW

**Comment by richhollis** [comment]
Hi @SamSaffron 

I've just pushed some revisions. Please take a look and let me know if this aligns with what you're thinking. If you need anything else then just let me know.

Cheers

File: lib/mini_profiler/profiler.rb [hunk 1]
```diff
@@ -127,6 +127,11 @@ def user(env)
       @config.user_provider.call(env)
     end
 
+    def content_security_policy_nonce(env)
+      config_csp_nonce = @config.content_security_policy_nonce
+      config_csp_nonce.is_a?(String) ? config_csp_nonce : config_csp_nonce.call(env)
```
Comment (SamSaffron) [inline lib/mini_profiler/profiler.rb:132]: I would prefer to "ultra" optimize this pipeline, can we wipe out the proc here? no need to have this extra indirection here imo: 

A simple thing like this should do the trick, no? 

`@config.content_security_policy_nonce || env["action_dispatch.content_security_policy_nonce"] || env["secure_headers_content_security_policy_nonce"]`

**Comment by SamSaffron** [comment]
looks great to me! thanks :confetti_ball:

**Comment by richhollis** [comment]
Great! :+1: This should close #439.

**Pull request merged**